### DISCUSSION TITLE
counts from database

### DISCUSSION THREAD

**Discussion by DevrayePatil** [Budibase/budibase#7054] [Help]
how can I get counts(how many results are present in the database) and conditional counts from the budibase database

**Comment by R2bEEaton** [comment]
You can do this a few different ways, here's just a couple.

### 1. Rows Length
This is the easiest way to count number of results if that's all you want to do. If you make a New Data Provider, you can create a paragraph or something underneath it to display `New Data Provider.Rows Length`. Make sure the Data Provider is not limited, or else if the true value is more than that limit, you will only ever see the limit. This could be slow depending on how many rows you have especially if you don't plan to filter it.

### 2. Group
Here is another way to count rows in a table.

Go to the desired table in the Data page and select "Create View."
![image](https://user-images.githubusercontent.com/34921506/182648190-c1ac2269-59f0-4f9a-b7f9-88d81ef74bc8.png)
Select "Calculate" and enter the following information.
![image](https://user-images.githubusercontent.com/34921506/182648360-7262cc0c-6c76-4521-85bc-e5e75e392a26.png)

Since the Auto ID is always unique for each row, this will result in counting the number of rows. Now you can use this as a data provider to display the count. You can also look into [Views Documentation](https://docs.budibase.com/docs/views) to see what else you can do with them, if you want to group these counts by a certain other field, etc. You can't really "filter" as in a regular table, but you can achieve a filter-like result depending on what you do grouping.

Hope this helps, enjoy!
